### DISCUSSION TITLE
GeoParquet output

### DISCUSSION THREAD

**Discussion by Kodiologist** [duckdb/duckdb#14274] [Q&A] [answered]
[Since 1.1.0](https://duckdb.org/2024/09/09/announcing-duckdb-110.html#geoparquet), DuckDB can read GeoParquet files with the `spatial` extension. But how can I produce GeoParquet output? I don't see GeoParquet or Parquet listed in `ST_Drivers()`, even though I'm using GDAL 3.8.4, which [should support Parquet](https://gdal.org/en/latest/drivers/vector/parquet.html).

If GeoParquet output is currently unimplemented, can you recommend a supported compressed geospatial file format, which is appropriate for large datasets? GeoJSON etc. are cute, but massively inflated for large datasets.

**Comment by Maxxen** [comment, marked as answer]
If you have the parquet extension and the spatial extension loaded, exporting normal parquet files containing `GEOMETRY` columns will automatically write the required geoparquet metadata to the output files. I.e. you don't have to do anything special, just export as parquet normally.

**Reply by Kodiologist** [reply]
Cool, thanks. I notice that when I read the result in R via Arrow (and `geoarrow`), no CRS is detected. Do I need to do anything special on DuckDB's end to make sure the CRS is preserved? If not, I guess I just have an unrelated R problem.

I used this SQL script to produce a GeoParquet file from Overture:

```sql
load spatial;
load httpfs;
SET s3_region='us-west-2';

copy(
    select
        id,
            names.primary as primary_name,
            geometry
        from read_parquet(
            's3://overturemaps-us-west-2/release/2024-09-18.0/theme=buildings/type=building/*',
            filename = true, hive_partitioning = 1)
        where
            bbox.xmin between -71.48 and -71.37 and
            bbox.ymin between 41.77 and 41.87)
    to '/tmp/build.parquet'
    with (format 'parquet');
```

Then I ran this in R:

```
> requireNamespace("geoarrow")
Loading required namespace: geoarrow
> library(sf)
Linking to GEOS 3.12.1, GDAL 3.8.4, PROJ 9.4.0; sf_use_s2() is TRUE
> head(st_as_sf(arrow::read_parquet("/tmp/build.parquet")))
Simple feature collection with 6 features and 2 fields
Geometry type: POLYGON
Dimension:     XY
Bounding box:  xmin: -71.48 ymin: 41.77008 xmax: -71.47881 ymax: 41.77212
CRS:           NA
                                id primary_name                       geometry
1 08b2a33029564fff02006093eaf1ae91         <NA> POLYGON ((-71.48 41.77154, ...
2 08b2a33029cddfff02002b9b667caa66         <NA> POLYGON ((-71.47979 41.772,...
3 08b2a33029193fff0200178aa2c153d2         <NA> POLYGON ((-71.47895 41.7702...
4 08b2a3302919efff0200357b3e74a634         <NA> POLYGON ((-71.47881 41.7701...
5 08b2a33029193fff020060666519fa92         <NA> POLYGON ((-71.47894 41.7703...
6 08b2a33029565fff0200f8eec0786bb5         <NA> POLYGON ((-71.47959 41.7709...
```

**Reply by Maxxen** [reply]
DuckDB does not track the CRS of geometry columns internally, so we don't output any. It's on my todo list to add an extra export option so that you can specify it manually.

**Reply by Maxxen** [reply]
Tracking CRS is also on the roadmap, but its going to require some planning to get right (and probably a lot of work as well) so I can't give a timeline for that just yet.

**Reply by Kodiologist** [reply]
I see. This is an important limitation, so perhaps it ought to be mentioned at https://duckdb.org/docs/extensions/spatial/overview.html . Maybe "the GEOMETRY type does not store `SRID` information on a per value basis" is meant to imply this. I tend to think of CRSes as applying to entire geometry sets rather than individual geometries.

**Reply by Maxxen** [reply]
Yes, that's the intention behind that remark in the docs. Even though DuckDB itself is commonly used as a pandas/dataframe alternative, I think the obvious comparison to the spatial extension (and spatial SQL in general) is PostGIS, which does store the CRS/SRID inside each individual geometry, allowing mixed-crs geometry columns.

**Reply by ar-siddiqui** [reply]
So, is it fair to say that as of now that the Geoparquet that DuckDB will write, will not have a CRS?

**Reply by Maxxen** [reply]
Yes, it writes the default CRS (wgs84)

**Reply by ar-siddiqui** [reply]
Thanks for replying, and what about the `bbox` column. It doesn't do that yet as well?
